pkg/aws/ec2.go

```diff
@@ -0,0 +1,62 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/smithy-go"
+)
+
+type EC2Client interface {
+	DescribeImages(ctx context.Context, params *ec2.DescribeImagesInput, optFns ...func(*ec2.Options)) (*ec2.DescribeImagesOutput, error)
+	DescribeKeyPairs(ctx context.Context, params *ec2.DescribeKeyPairsInput, optFns ...func(*ec2.Options)) (*ec2.DescribeKeyPairsOutput, error)
+	CreateKeyPair(ctx context.Context, params *ec2.CreateKeyPairInput, optFns ...func(*ec2.Options)) (*ec2.CreateKeyPairOutput, error)
+}
+
+func NewEC2Client(config aws.Config) *ec2.Client {
+	return ec2.NewFromConfig(config)
+}
+
+func (a *Client) ImageExists(ctx context.Context, imageID string) (bool, error) {
+	params := &ec2.DescribeImagesInput{
+		ImageIds: []string{imageID},
+	}
+	_, err := a.ec2.DescribeImages(ctx, params)
+	if err == nil {
+		return true, nil
+	}
+
+	var apiErr smithy.APIError
+	if errors.As(err, &apiErr) && apiErr.ErrorCode() == "400" {
+		return false, nil
+	}
+	return false, fmt.Errorf("aws describe image [%s]: %v", imageID, err)
```

Comment:
This error results in an arbitrary hash that the user must assume represents an image ID. Easily cleared up with a key.

```suggestion
	return false, fmt.Errorf("aws describe image [imageID=%s]: %v", imageID, err)
```

We should also agree on whether errors use square or rounded parenthesis.